app/main.py

```diff
@@ -1,8 +1,40 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+
+        Person.people[name] = self
+
+    def set_spouse(self, spouse_name: str) -> str:
+
+        if spouse_name is None:
+            return
+
+        spouse = Person.people.get(spouse_name)
```

Comment:
There is no handling for the case where `spouse_name` does not exist in `Person.people`. Consider adding a check and returning an appropriate message if the spouse is not found.